### PR TITLE
Orders user userEmail performance

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it improves fields user and userEmail execution on orders query.
Related tickets:
[SALEOR-2733](https://app.clickup.com/t/2549495/SALEOR-2733)
[SALEOR-2734](https://app.clickup.com/t/2549495/SALEOR-2734)

Local testing(10 orders):
before: "query-count": 16
after: "query-count": 7

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [X] Privileged queries and mutations are guarded by proper permission checks
* [X] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [X] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2733 Optimize user resolver in Order type](https://app.clickup.com/t/2549495/SALEOR-2733)

**Comment by patrys** [comment]
Task linked: [SALEOR-2734 Optimize userEmail resolver in Order type](https://app.clickup.com/t/2549495/SALEOR-2734)

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/order/types.py [hunk 1]
```diff
@@ -692,25 +692,29 @@ def return_voucher_translated_name(discounts) -> Optional[Money]:
...
+        def _resolve_billing_address(data):
+            user, address = data
+            requester = get_user_or_app_from_context(info.context)
+            if requestor_has_access(requester, user, OrderPermissions.MANAGE_ORDERS):
+                return address
+            return obfuscate_address(address)
+
+        user = UserByUserIdLoader(info.context).load(root.user_id)
+        address = AddressByIdLoader(info.context).load(root.billing_address_id)
+        return Promise.all([user, address]).then(_resolve_billing_address)
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:704]: Please, make sure it's not failing when `billing_address` is `None`.

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -692,25 +692,29 @@ def return_voucher_translated_name(discounts) -> Optional[Money]:
...
+        def _resolve_shipping_address(data):
+            user, address = data
+            requester = get_user_or_app_from_context(info.context)
+            if requestor_has_access(requester, user, OrderPermissions.MANAGE_ORDERS):
+                return address
+            return obfuscate_address(address)
+
+        user = UserByUserIdLoader(info.context).load(root.user_id)
+        address = AddressByIdLoader(info.context).load(root.shipping_address_id)
+        return Promise.all([user, address]).then(_resolve_shipping_address)
```
Comment (IKarbowiak) [inline saleor/graphql/order/types.py:717]: Same here for `shipping_address`.

File: saleor/graphql/order/types.py [hunk 3]
```diff
@@ -692,29 +692,33 @@ def return_voucher_translated_name(discounts) -> Optional[Money]:
...
             return
-        requester = get_user_or_app_from_context(info.context)
-        if requestor_has_access(requester, root.user, OrderPermissions.MANAGE_ORDERS):
-            return AddressByIdLoader(info.context).load(root.billing_address_id)
-        return (
-            AddressByIdLoader(info.context)
-            .load(root.billing_address_id)
-            .then(obfuscate_address)
-        )
+        user = UserByUserIdLoader(info.context).load(root.user_id)
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:704]: This dataloader will fail if there is no user assigned to this dataloader.

File: saleor/graphql/order/types.py [hunk 4]
```diff
@@ -692,29 +692,33 @@ def return_voucher_translated_name(discounts) -> Optional[Money]:
...
             return
-        requester = get_user_or_app_from_context(info.context)
-        if requestor_has_access(requester, root.user, OrderPermissions.MANAGE_ORDERS):
-            return AddressByIdLoader(info.context).load(root.shipping_address_id)
-        return (
-            AddressByIdLoader(info.context)
-            .load(root.shipping_address_id)
-            .then(obfuscate_address)
-        )
+        user = UserByUserIdLoader(info.context).load(root.user_id)
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:719]: This dataloader will fail if there is no user assigned to this dataloader.

File: saleor/graphql/order/types.py (hunk 3 above)
Comment (fowczarek) [reply saleor/graphql/order/types.py:704]: We should add a similar test to `test_query_order_without_addresess`, but we should fetch the address for the order without the assigned user.

Comment (tomaszszymanski129) [reply saleor/graphql/order/types.py:704]: yep, i've seen that, added in_progress label to fix it and tweak the performance further

Comment (patrys) [reply saleor/graphql/order/types.py]: To make it clear: it's correct for the dataloader to fail, we should not call the loader if there is nothing to load (I don't want dataloaders to discard incorrect keys silently).

**Review by fowczarek — APPROVED** [review]

**Pull request merged**